cns/api.go

```diff
@@ -48,8 +48,17 @@ type HTTPService interface {
 	GetPendingReleaseIPConfigs() []IPConfigurationStatus
 	GetPodIPConfigState() map[string]IPConfigurationStatus
 	MarkIPAsPendingRelease(numberToMark int) (map[string]IPConfigurationStatus, error)
+	AttachSWIFTv2Middleware(middleware SWIFTv2Middleware)
```

Comment:
do we need this in the service interface? seems like an implementation detail

Reply:
yeah, because in `InitializeCRDState` we are atually passing in the `HTTPService` interface, and then convert that interface type to implementation type.